ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -91,6 +95,14 @@ data LedgerInterface m blk = LedgerInterface
   { getCurrentLedgerState :: STM m (LedgerState blk)
   }
 
+-- | Represents the maximum number of transactions that a 'Mempool' can
+-- contain.
+--
+-- TODO @intricate: should this be a data type that also allows for an
+-- uncapped mempool?
+-- e.g. data MempoolCapacity = MempoolCapacity !Word64 | NoMempoolCapacity
```

Comment:
I don't think it'd be any more difficult to limit the mempool in bytes since we have the `txSize :: GenTx blk -> TxSizeInBytes` function.

Reply:
I'll open an issue for re-implementing `MempoolCapacity` as a byte limit rather than number of transactions. I think I'd like to get this functionality merged first and then take a look at this.

Reply:
Opened issue: https://github.com/input-output-hk/ouroboros-network/issues/974